Partitioned heat conduction (G+Smo participant) (#604)
precice/tutorials: partitioned-heat-conduction/README.md

* OpenFOAM. This case uses the custom [heatTransfer](https://github.com/precice/tutorials/blob/master/partitioned-heat-conduction/solver-openfoam/heatTransfer.C) solver (find it in `solver-openfoam` and build with `wmake`). Read more details in the [OpenFOAM adapter](https://precice.org/adapter-openfoam-overview.html).

> * G+Smo. Inatsll [G+Smo](https://github.com/gismo/gismo).

Reviewer: ```suggestion
* G+Smo. Install [G+Smo](https://github.com/gismo/gismo).
```

But we need a bit more details, or better link to the adapter documentation:

```suggestion
* G+Smo. Install the [G+Smo adapter](https://precice.org/adapter-gismo.html).
```

mpirun -n <N_PROC> heat.py -d
```

The G+Smo-based version of the tutorial offers IsoGeometric Analysis discretization method. To run the example you need to follow the following steps:

- Download G+Smo and Create a Build Folder
```
git clone git@github.com:gismo/gismo.git
cd gismo

> mkdir build

Reviewer: ```suggestion
cd gismo
mkdir build
cd build
```

```
 - Configure G+Smo 
```

> cmake .. -DGISMO_OPTIONAL="<other submodules>;gsPreCICE"

Reviewer: Do we need other modules for this tutorial? If not, let's remove. If yes, document.

But as I wrote in another comment, this looks like something to move to the adapter documentation.

```
- Build the Example
```

> make partitioned-heat-conduction -j <number of threads to use>

Reviewer: This is still failing for me with:

```
commit 62078b48516c4a80869c56caab83c393f13640e8 (HEAD -> stable, origin/stable, origin/HEAD)
Merge: cac651c3 f2565ee4
Author: Hugo Verhelst <38376601+hverhelst@users.noreply.github.com>
Date:   Wed Dec 17 15:31:48 2025 +0100

    Merge pull request #778 from gismo/gsCahnHilliardAssembler
    
    Adding a black-box assembler for the Cahn-Hilliard equation & Color output
```

Here is the full log: [gismo-building-issue.log](https://github.com/user-attachments/files/24532037/gismo-building-issue.log)

Reviewer: This worked with gsPreCICE commit e557b51.

```
- Link the compiled executable to the gismo-executable folder within the tutorial directory
```
cd <Your preCICE tutorial folder>/partitioned-heat-conduction/gismo-executable
ln -sf <You G+Smo build folder>/bin/partitioned-heat-conduction ./gismo_executable`
```
- Open two terminals and run
```
 cd dirichlet-gismo
 ./run.sh
```

```
cd neumann-gismo
./run.sh

> ```

Reviewer: Could we move most of this content to the adapter documentation, and refer to that?

The path `./gismo_executable/gismo_executable` is assumed by the `run.sh`, so it should probably be documented here. But we should better call that directory `solver-gismo`.

mpirun -n <N_PROC> heat.py -d
```

For running G+Smo, follow the [G+Smo adapter installation instructions](https://precice.org/adapter-gismo.html) to build the `partitioned-heat-conduction` example. Then link the compiled executable to the `dirichlet-gismo` and `neumann-gismo` folders:

```bash
cd <tutorial-directory>/partitioned-heat-conduction/dirichlet-gismo
ln -sf <gismo-build-directory>/bin/partitioned-heat-conduction ./gismo-executable
cd ../neumann-gismo
ln -sf <gismo-build-directory>/bin/partitioned-heat-conduction ./gismo-executable
```

Then open two terminals and run:

```bash
cd dirichlet-gismo
./run.sh
```

```bash
cd neumann-gismo

> ./run.sh

Reviewer: While I can run the `partitioned-heat-conduction-direct` case, this case gives me a configuration error early on. Do I need a different configuration or solver executable?

```
root@dfc17ae89aca:/mnt/host/tutorials/partitioned-heat-conduction/neumann-gismo# ./run.sh 
Started on: Wed, 13 May 2026 13:35:46 +0200
Patches: 1, degree: 2
Active options:
Options (12):
* DirichletStrategy    (int) = 11       Method for enforcement of Dirichlet BCs [11..14]
* DirichletValues      (int) = 101      Method for computation of Dirichlet DoF values [100..103]
* SameElement         (bool) = 1        Activates optimization if all quadrature points are located in the same element
* bdA                 (real) = 2        Estimated nonzeros per column of the matrix: bdA*deg + bdB
* bdB                  (int) = 1        Estimated nonzeros per column of the matrix: bdA*deg + bdB
* bdO                 (real) = 0.333    Overhead of sparse mem. allocation: (1+bdO)(bdA*deg + bdB) [0..1]
* flipSide            (bool) = 0        Flip side of interface where integration is performed.
* movingInterface     (bool) = 0        Used in interface assembly when interface is not stationary.
* overInt             (bool) = 0        Apply over-integration on boundary elements or not?
* quA                 (real) = 1        Number of quadrature points: quA*deg + quB; For patchRule: Regularity of the target space
* quB                  (int) = 1        Number of quadrature points: quA*deg + quB; For patchRule: Degree of the target space
* quRule               (int) = 1        Quadrature rule used (1) Gauss-Legendre; (2) Gauss-Lobatto; (3) Patch-Rule

---[precice]  This is preCICE version 3.4.1
---[precice]  Revision info: no-info [git failed to run]
---[precice]  Build type: Release (without debug log)
---[precice]  Working directory "/mnt/host/tutorials/partitioned-heat-conduction/neumann-gismo"
---[precice]  Configuring preCICE with configuration "../precice-config.xml"
---[precice]  I am participant "Neumann"
GISMO_DEBUG: gsPreCICE.h:178, "setMeshVertices": 
setMeshVertices
---[precice] ERROR:  This participant attempteded to set an access region (via "setMeshAccessRegion") on mesh "Dirichlet-Mesh", but mesh "Dirichlet-Mesh" is either not a received mesh or its api access was not enabled in the configuration. setMeshAccessRegion(...) is only valid for (<receive-mesh name="Dirichlet-Mesh" ... api-access="true"/>).
terminate called after throwing an instance of 'precice::Error'
  what():  This participant attempteded to set an access region (via "setMeshAccessRegion") on mesh "Dirichlet-Mesh", but mesh "Dirichlet-Mesh" is either not a received mesh or its api access was not enabled in the configuration. setMeshAccessRegion(...) is only valid for (<receive-mesh name="Dirichlet-Mesh" ... api-access="true"/>).
```

For Nutils, please use the files `Dirichlet-*.vtk` or `Neumann-*.vtk`. Please note that these files contain the temperature as well as the reference solution.

> For G+Smo, please use the file `solution.pvd` in both dirichlet-gismo and neumann-gismo directories. 

Reviewer: ```suggestion
For G+Smo, please use the generated `solution.pvd` files in the dirichlet-gismo and neumann-gismo directories. 
```

(I thought this was a file in this PR for a moment)
